backend/src/main/java/org/compilers/cryptoyard/messages/MessageDispatcher.java

```diff
@@ -0,0 +1,44 @@
+package org.compilers.cryptoyard.messages;
+
+import org.compilers.cryptoyard.services.CYService;
+import org.springframework.stereotype.Component;
+
+import java.util.Set;
+
+/**
+ * Responsible for managing subscriptions and forwarding messages to necessary services.
+ */
+@Component
+public class MessageDispatcher {
+    private final Subscriptions subscriptions = new Subscriptions();
+    /**
+     * Register listener for a specific topic. All messages of the current topic will be forwarded to the listener
+     * @param topic Topic to subscribe to - a message class
+     * @param subscriber The service to which all the messages on the given topic will be forwarded
+     * @param <C> A subclass for Message
```

Comment:
@return instead of @param?

Reply:
The return type is void. IntelliJ suggested it as a param. Removed that to avoid confusion.